File: molecular/src/main/java/com/hartwig/actin/molecular/orange/evidence/matching/GeneMatching.java
package com.hartwig.actin.molecular.orange.evidence.matching;

import com.hartwig.hmftools.datamodel.purple.PurpleVariant;
import com.hartwig.serve.datamodel.MutationType;
import com.hartwig.serve.datamodel.gene.GeneAnnotation;

import org.jetbrains.annotations.NotNull;

public final class GeneMatching {

    private GeneMatching() {
    }

    public static boolean isMatch(@NotNull GeneAnnotation geneAnnotation, @NotNull PurpleVariant variant) {
        boolean geneMatch = geneAnnotation.gene().equals(variant.gene());
        boolean typeMatch = MutationTypeMatching.matches(MutationType.ANY, variant);

        return geneMatch && typeMatch;
    }
}

File: molecular/src/main/kotlin/com/hartwig/actin/molecular/orange/evidence/matching/GeneMatching.kt
package com.hartwig.actin.molecular.orange.evidence.matching

import com.hartwig.hmftools.datamodel.purple.PurpleVariant
import com.hartwig.serve.datamodel.MutationType
import com.hartwig.serve.datamodel.gene.GeneAnnotation

object GeneMatching {

    fun isMatch(geneAnnotation: GeneAnnotation, variant: PurpleVariant): Boolean {
        val geneMatch = geneAnnotation.gene() == variant.gene()
        val typeMatch = MutationTypeMatching.matches(MutationType.ANY, variant)
        return geneMatch && typeMatch
    }
}
